### ISSUE TITLE
Unrecognized selector crash

### ISSUE THREAD

**Issue by spnkr** [Alamofire/Alamofire#3432]
* [ ] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines](https://github.com/Alamofire/Alamofire/blob/master/CONTRIBUTING.md).

## What did you do?

I have multiple NSManagedObjectContexts, and am calling `AlamoFireSession.perform(DataRequest)` from inside them and from the main thread.

This means that `AlamoFireSession.perform(DataRequest)` is being called from more than one thread. Could this cause the crash?

## What did you expect to happen?

I expected no crash. 

## What happened instead?

Got crash on thread 14 `NSInvalidArgumentException: -[__NSTaggedDate member:]: unrecognized selector sent to instance 0x8000000000000000`

```
Thread 14
0  CoreFoundation          ___exceptionPreprocess
1  libobjc.A.dylib         _objc_exception_throw
2  CoreFoundation          -[NSObject(NSObject) doesNotRecognizeSelector:]
3  CoreFoundation          ____forwarding___
4  CoreFoundation          ___forwarding_prep_0___
5  libswiftCore.dylib      Swift.Set.insert(__owned A) -> (inserted: Swift.Bool, memberAfterInsert: A)
6  Example App             closure #1 () -> () in AlamoFireSession.perform(DataRequest) -> () (AlamoFireSession.swift:826:33)
7  <compiler-generated>    reabstraction thunk helper from @escaping @callee_guaranteed () -> () to @escaping @callee_unowned @convention(block) () -> ()
8  libdispatch.dylib       __dispatch_call_block_and_release
9  libdispatch.dylib       __dispatch_client_callout
10 libdispatch.dylib       __dispatch_lane_serial_drain
11 libdispatch.dylib       __dispatch_lane_invoke
12 libdispatch.dylib       __dispatch_workloop_worker_thread
13 libsystem_pthread.dylib __pthread_wqthread
14 libsystem_pthread.dylib _start_wqthread
```

## Alamofire Environment

**Alamofire Version:** 5.0.2
**Dependency Manager:** None
**Xcode Version:** 12.4
**Swift Version:** 5.3
**Platform(s) Running Alamofire:**
**macOS Version Running Xcode:** 10.15.7
**iOS Version:** 14.4.1
**iPhone type:** iPhone12,3 (iPhone 11 Pro)

**Comment by spnkr** [comment]
Since AlamoFireSession.perform(DataRequest) is being called from more than one thread, the only idea I have is this:

Wherever I do this now:

```
AlamoFireSession.default.request(...)
```

I replace it with this:

```
let queue = DispatchQueue(label:"all-requests-here")

// and then elsewhere
queue.async {
    AlamoFireSession.default.request(...)
}
```

where the queue object is scoped so as to stick around.

**Comment by jshier** [comment]
I'm not sure what API you're using here, but it's not Alamofire's. Alamofire has no `AlamoFireSession` type. I would suggest updating Alamofire to the latest version, 5.4.2, and then using the Thread Sanitizer to identify any possible thread-safety issues you might be seeing. However, from the look of the crash stack and exception, Alamofire itself isn't involved. I'll leave this issue open for a few days while you investigate.

**Comment by jshier** [comment]
I saw your Stack Overflow. If this is `activeRequests`, that thread-safety issue was fixed in [5.2.0](https://github.com/Alamofire/Alamofire/pull/3179).
